nmt.go

```diff
@@ -304,6 +365,7 @@ func (n *NamespacedMerkleTree) Push(namespacedData namespace.PrefixedData) error
 
 	// update relevant "caches":
 	n.leaves = append(n.leaves, namespacedData)
+	// TODO [Me] why not calculating the hash right away and add it to n.leafHashes
```

Comment:
I do not remember if there was any reason for not doing that. Maybe the benchmarks where favourable for hashing all at the end? Though I do not see how that could be more efficient at all.